Can no longer compile without `deprecated` flag

### Godot version

master

### System information

Ubuntu

### Issue description

I think one of the recent changes wrapped the project converter in `#ifndef DISABLE_DEPRECATED` but missed the coresponding code in `main.cpp` (and potentially other places).
```
main/main.cpp: In static member function 'static bool Main::start()':
main/main.cpp:2551:33: error: 'ProjectConverter3To4' was not declared in this scope
 2551 |                 int exit_code = ProjectConverter3To4(converter_max_kb_file, converter_max_line_length).convert();
      |                                 ^~~~~~~~~~~~~~~~~~~~
main/main.cpp:2556:33: error: 'ProjectConverter3To4' was not declared in this scope
 2556 |                 int exit_code = ProjectConverter3To4(converter_max_kb_file, converter_max_line_length).validate_conversion();
```

### Steps to reproduce

for instance, try to compile with `deprecated = "no"` in your `custom.py`.

### Minimal reproduction project

n/a

--- a/main/main.cpp
+++ b/main/main.cpp
@@ -2366,8 +2366,10 @@ bool Main::start() {
 	String _export_preset;
 	bool export_debug = false;
 	bool export_pack_only = false;
+#ifndef DISABLE_DEPRECATED
 	bool converting_project = false;
 	bool validating_converting_project = false;
+#endif // DISABLE_DEPRECATED
 #endif
 
 	main_timer_sync.init(OS::get_singleton()->get_ticks_usec());
@@ -2383,10 +2385,12 @@ bool Main::start() {
 #ifdef TOOLS_ENABLED
 		} else if (args[i] == "--no-docbase") {
 			doc_base = false;
+#ifndef DISABLE_DEPRECATED
 		} else if (args[i] == "--convert-3to4") {
 			converting_project = true;
 		} else if (args[i] == "--validate-conversion-3to4") {
 			validating_converting_project = true;
+#endif // DISABLE_DEPRECATED
 		} else if (args[i] == "-e" || args[i] == "--editor") {
 			editor = true;
 		} else if (args[i] == "-p" || args[i] == "--project-manager") {
@@ -2547,6 +2551,7 @@ bool Main::start() {
 		return false;
 	}
 
+#ifndef DISABLE_DEPRECATED
 	if (converting_project) {
 		int exit_code = ProjectConverter3To4(converter_max_kb_file, converter_max_line_length).convert();
 		OS::get_singleton()->set_exit_code(exit_code);
@@ -2557,6 +2562,7 @@ bool Main::start() {
 		OS::get_singleton()->set_exit_code(exit_code);
 		return false;
 	}
+#endif // DISABLE_DEPRECATED
 
 #endif
 

